edb/schema/inheriting.py

```diff
@@ -98,10 +98,15 @@ def inherit_fields(
         inherited_fields_update = {}
         deferred_complex_ops = []
 
-        for field_name in field_names:
+        # Iterate over mcls.get_schema_fields() instead of field_names for
```

Comment:
Why is this important?

Reply:
Compilation of the trigger expression depends on the values of `kind` and `scope`, and so we need to arrange to inherit those before we try to inherit `expr`